mkosi/__init__.py

```diff
@@ -1752,6 +1752,41 @@ def clean_tdnf_metadata(root: Path, always: bool) -> None:
             unlink_try_hard(path)
 
 
+def clean_apt_metadata(root: Path, always: bool) -> None:
+    """Remove apt metadata if /usr/bin/apt is not present in the image"""
+    paths = [
+        root / "var/lib/apt",
+        root / "var/log/apt",
+        root / "var/cache/apt",
+    ]
+
+    cond = always or os.access(root / "usr/bin/apt", os.F_OK, follow_symlinks=False)
```

Comment:
This is copied from the other implementations, but I'm not sure I understand - it deletes the metadata if the executables are installed? Shouldn't it be the opposite?

Reply:
Oops, the condtion got reversed again. Maybe fix them all first?